test/e2e/apps/statefulset.go

```diff
@@ -510,6 +512,145 @@ var _ = SIGDescribe("StatefulSet", func() {
 
 		})
 
+		ginkgo.It("should perform canary updates and phased rolling updates of template modifications for partiton1 and delete pod-0", func(ctx context.Context) {
```

Comment:
IIUC the test in the current version passes on master because the deleted pod-0 ends in Succeeded phase, and this is not recreated by StatefulSet controller because https://github.com/kubernetes/kubernetes/pull/120398 got reverted. 

Because testing both scenarios makes sense (one reproduces now, the other will prevent regression related to the other PR), I suggest to make two versions of the test, one with basic ss, the other with extra failing container. For example: 
```golang
	ss.Spec.Template.Spec.Containers = append(ss.Spec.Template.Spec.Containers, v1.Container{
		Name:    "sleep-exit-with-1",
		Image:   imageutils.GetE2EImage(imageutils.BusyBox),
		Command: []string{"sh", "-c"},
		Args: []string{`
                                  _term(){
                                     echo "Received SIGTERM signal"
                                     exit 1
                                  }
                                  trap _term SIGTERM
                                  while true; do
                                     echo "Running..."
                                     sleep 1
                                  done
			`,
		},
	})
```

One way to commonize the code test it to wrap the test body in a function (parametrized with ss and potentially other stuff), similarly as it is done with the use of `rollbackTest`.